internal/signing/channelaccounts/channel_accounts_model_test.go

```diff
@@ -0,0 +1,211 @@
+	dbt := dbtest.Open(t)
+	defer dbt.Close()
+
+	dbConnectionPool, err := db.OpenDBConnectionPool(dbt.DSN)
+	require.NoError(t, err)
+	defer dbConnectionPool.Close()
+
+	ctx := context.Background()
+	m := NewChannelAccountModel(dbConnectionPool)
+
+	channelAccount := keypair.MustRandom()
+	ca, err := m.Get(ctx, dbConnectionPool, channelAccount.Address())
+	assert.ErrorIs(t, err, ErrChannelAccountNotFound)
+	assert.Nil(t, ca)
+
+	createChannelAccountFixture(t, ctx, dbConnectionPool, channelAccount.Address(), channelAccount.Seed())
+	ca, err = m.Get(ctx, dbConnectionPool, channelAccount.Address())
+	require.NoError(t, err)
+	assert.Equal(t, ca.PublicKey, channelAccount.Address())
+	assert.Equal(t, ca.EncryptedPrivateKey, channelAccount.Seed())
+}
+
+func TestChannelAccountModelGetAllByPublicKey(t *testing.T) {
+	dbt := dbtest.Open(t)
+	defer dbt.Close()
+
+	dbConnectionPool, err := db.OpenDBConnectionPool(dbt.DSN)
+	require.NoError(t, err)
+	defer dbConnectionPool.Close()
+
+	ctx := context.Background()
+	m := NewChannelAccountModel(dbConnectionPool)
+
+	channelAccount1 := keypair.MustRandom()
+	channelAccount2 := keypair.MustRandom()
+	createChannelAccountFixture(t, ctx, dbConnectionPool, channelAccount1.Address(), channelAccount1.Seed())
+	createChannelAccountFixture(t, ctx, dbConnectionPool, channelAccount2.Address(), channelAccount2.Seed())
+
+	channelAccounts, err := m.GetAllByPublicKey(ctx, dbConnectionPool, channelAccount1.Address(), channelAccount2.Address())
+	require.NoError(t, err)
+
+	assert.Len(t, channelAccounts, 2)
+	assert.Equal(t, channelAccount1.Address(), channelAccounts[0].PublicKey)
+	assert.Equal(t, channelAccount2.Address(), channelAccounts[1].PublicKey)
+}
+
+func TestChannelAccountModelBatchInsert(t *testing.T) {
+	dbt := dbtest.Open(t)
+	defer dbt.Close()
+
+	dbConnectionPool, err := db.OpenDBConnectionPool(dbt.DSN)
+	require.NoError(t, err)
+	defer dbConnectionPool.Close()
+
+	ctx := context.Background()
+	m := NewChannelAccountModel(dbConnectionPool)
+
+	t.Run("channel_accounts_empty", func(t *testing.T) {
+		err := m.BatchInsert(ctx, dbConnectionPool, []*ChannelAccount{})
+		require.NoError(t, err)
+	})
+
+	t.Run("invalid_channel_accounts", func(t *testing.T) {
+		channelAccounts := []*ChannelAccount{
+			{
+				PublicKey: "",
+			},
+		}
+		err := m.BatchInsert(ctx, dbConnectionPool, channelAccounts)
+		assert.EqualError(t, err, "public key cannot be empty")
+
+		channelAccounts = []*ChannelAccount{
+			{
+				PublicKey: keypair.MustRandom().Address(),
+			},
+		}
+		err = m.BatchInsert(ctx, dbConnectionPool, channelAccounts)
+		assert.EqualError(t, err, "private key cannot be empty")
+	})
+
+	t.Run("inserts_channel_accounts_successfully", func(t *testing.T) {
+		channelAccount1 := keypair.MustRandom()
+		channelAccount2 := keypair.MustRandom()
+		channelAccounts := []*ChannelAccount{
+			{
+				PublicKey:           channelAccount1.Address(),
+				EncryptedPrivateKey: channelAccount1.Seed(),
```

Comment:
Perhaps would be interesting to check on `BatchInsert` that the private key is not in its raw format but actually encrypted, wdyt?

Reply:
This logic shouldn't be in the model layer.